src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServer.java

```diff
@@ -119,8 +119,15 @@ public static boolean RequestHTTPS(
 
       if (fileStr.charAt(0) != '[' && fileStr.charAt(0) != '{') {
         // json has odd characters at the front, trim them to make valid json
-        int i = fileStr.indexOf("{");
-        fileStr = fileStr.substring(i);
+        int jsonStartIndex = fileStr.indexOf("{");
+
+        int notFound = -1;
+        if (jsonStartIndex == notFound) {
+          connected = true;
+          Logger.LOG_SERIOUS("Did not receive a valid response from server!");

```

Comment:
it might be clear if you specify osisoft server here. Also you may want to rephrase to something like "Received response from OSIsoft server was not valid". The way you have it, it could be read that you just didnt receive a response or the response received was invalid.

Are you logging invalid responses? They might be meaningful to troubleshooting.

Reply:
Right now those responses are saved to the file system which isn't great. I'm working on changing this to log out any error messages or invalid responses. It won't be in this PR.